wire-library/wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java

```diff
@@ -218,8 +222,8 @@ public static boolean builtInType(ProtoType protoType) {
           String suggestion = collidingNames.contains(field.getName())
             || (field.getName().equals(field.getType().getSimpleName())
               && !field.getType().isScalar())
-            || schema.getType(field.getQualifiedName()) != null
-              ? field.getQualifiedName()
+            || hasEponymousType(schema, field)
+              ? legacyQualifiedFieldName(field)
```

Comment:
This effectively reverts the prior changes in this PR that resulted in differences in generated code for when a field is decided to conflict with a type name.